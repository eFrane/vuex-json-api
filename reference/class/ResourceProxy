<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Json:Api for Vuex</title>
    <meta name="generator" content="VuePress 1.8.0">
    
    <meta name="description" content="Automagic Json:Api integration for Vuex">
    
    <link rel="preload" href="/assets/css/0.styles.ba26daf8.css" as="style"><link rel="preload" href="/assets/js/app.c7d15b50.js" as="script"><link rel="preload" href="/assets/js/2.0dafffee.js" as="script"><link rel="preload" href="/assets/js/47.c5850462.js" as="script"><link rel="prefetch" href="/assets/js/10.6f689e45.js"><link rel="prefetch" href="/assets/js/11.3e854a07.js"><link rel="prefetch" href="/assets/js/12.ed1a8014.js"><link rel="prefetch" href="/assets/js/13.4b10f9ae.js"><link rel="prefetch" href="/assets/js/14.b4e56768.js"><link rel="prefetch" href="/assets/js/15.2c72dc07.js"><link rel="prefetch" href="/assets/js/16.5963f83e.js"><link rel="prefetch" href="/assets/js/17.20955945.js"><link rel="prefetch" href="/assets/js/18.9338e718.js"><link rel="prefetch" href="/assets/js/19.d21db1cb.js"><link rel="prefetch" href="/assets/js/20.e524e9be.js"><link rel="prefetch" href="/assets/js/21.47a4ed11.js"><link rel="prefetch" href="/assets/js/22.2cb1dca4.js"><link rel="prefetch" href="/assets/js/23.7c076f99.js"><link rel="prefetch" href="/assets/js/24.6ac270b7.js"><link rel="prefetch" href="/assets/js/25.4f939962.js"><link rel="prefetch" href="/assets/js/26.8ae51a44.js"><link rel="prefetch" href="/assets/js/27.e2a59626.js"><link rel="prefetch" href="/assets/js/28.cdd1dd4b.js"><link rel="prefetch" href="/assets/js/29.4e9281f4.js"><link rel="prefetch" href="/assets/js/3.437a0d76.js"><link rel="prefetch" href="/assets/js/30.b629fffc.js"><link rel="prefetch" href="/assets/js/31.e800fa24.js"><link rel="prefetch" href="/assets/js/32.0e5c08fb.js"><link rel="prefetch" href="/assets/js/33.228977a9.js"><link rel="prefetch" href="/assets/js/34.eacfc929.js"><link rel="prefetch" href="/assets/js/35.bc16c9ad.js"><link rel="prefetch" href="/assets/js/36.329aa466.js"><link rel="prefetch" href="/assets/js/37.5da81bb5.js"><link rel="prefetch" href="/assets/js/38.d343cd87.js"><link rel="prefetch" href="/assets/js/39.d3a1fd17.js"><link rel="prefetch" href="/assets/js/4.bfe88385.js"><link rel="prefetch" href="/assets/js/40.0bbfd7d9.js"><link rel="prefetch" href="/assets/js/41.507bbe77.js"><link rel="prefetch" href="/assets/js/42.9049d9e3.js"><link rel="prefetch" href="/assets/js/43.1e11c805.js"><link rel="prefetch" href="/assets/js/44.9b865ea6.js"><link rel="prefetch" href="/assets/js/45.a443ddf5.js"><link rel="prefetch" href="/assets/js/46.c7c502ac.js"><link rel="prefetch" href="/assets/js/48.14e53295.js"><link rel="prefetch" href="/assets/js/49.1c934608.js"><link rel="prefetch" href="/assets/js/5.cb249b63.js"><link rel="prefetch" href="/assets/js/50.91872906.js"><link rel="prefetch" href="/assets/js/51.b9cfcbed.js"><link rel="prefetch" href="/assets/js/52.4aa89f85.js"><link rel="prefetch" href="/assets/js/53.7089bb5b.js"><link rel="prefetch" href="/assets/js/54.cd3cd80c.js"><link rel="prefetch" href="/assets/js/55.01aab259.js"><link rel="prefetch" href="/assets/js/56.0161cdb4.js"><link rel="prefetch" href="/assets/js/57.69ed9460.js"><link rel="prefetch" href="/assets/js/58.4051217a.js"><link rel="prefetch" href="/assets/js/59.becdceb6.js"><link rel="prefetch" href="/assets/js/6.4155846f.js"><link rel="prefetch" href="/assets/js/60.9977140e.js"><link rel="prefetch" href="/assets/js/61.074d1cf4.js"><link rel="prefetch" href="/assets/js/62.25ed9d76.js"><link rel="prefetch" href="/assets/js/63.b9031c84.js"><link rel="prefetch" href="/assets/js/64.74f38c3d.js"><link rel="prefetch" href="/assets/js/65.fe7e5276.js"><link rel="prefetch" href="/assets/js/7.6ed4aaa9.js"><link rel="prefetch" href="/assets/js/8.58c932af.js"><link rel="prefetch" href="/assets/js/9.9a6f2cf9.js">
    <link rel="stylesheet" href="/assets/css/0.styles.ba26daf8.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container no-sidebar"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><!----> <span class="site-name">Json:Api for Vuex</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <!----> </aside> <main class="page"> <div class="theme-default-content content__default"><p><a name="ResourceProxy"></a></p> <h2 id="resourceproxy"><a href="#resourceproxy" class="header-anchor">#</a> ResourceProxy</h2> <p><strong>Kind</strong>: global class</p> <ul><li><a href="#ResourceProxy">ResourceProxy</a> <ul><li><a href="#new_ResourceProxy_new">new ResourceProxy()</a></li> <li><a href="#ResourceProxy+addRoute">.addRoute(route)</a></li> <li><a href="#ResourceProxy+methodNotAvailable">.methodNotAvailable(methodName)</a></li> <li><a href="#ResourceProxy+get">.get(parameters, data)</a> ⇒</li> <li><a href="#ResourceProxy+list">.list(parameters, data)</a> ⇒</li> <li><a href="#ResourceProxy+create">.create(parameters, data)</a> ⇒</li> <li><a href="#ResourceProxy+replace">.replace(parameters, data)</a> ⇒</li> <li><a href="#ResourceProxy+update">.update(parameters, data)</a> ⇒</li> <li><a href="#ResourceProxy+delete">.delete(parameters, data)</a> ⇒</li></ul></li></ul> <p><a name="new_ResourceProxy_new"></a></p> <h3 id="new-resourceproxy"><a href="#new-resourceproxy" class="header-anchor">#</a> new ResourceProxy()</h3> <p>ResourceProxy</p> <p>A wrapper class for the available methods on
a route. The actual proxied axios-clients
will only be created once the first request on
a method is made.</p> <p><a name="ResourceProxy+addRoute"></a></p> <h3 id="resourceproxy-addroute-route"><a href="#resourceproxy-addroute-route" class="header-anchor">#</a> resourceProxy.addRoute(route)</h3> <p>Adds a method to the resource</p> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>route</td> <td><code>Route</code></td></tr></tbody></table> <p><a name="ResourceProxy+methodNotAvailable"></a></p> <h3 id="resourceproxy-methodnotavailable-methodname"><a href="#resourceproxy-methodnotavailable-methodname" class="header-anchor">#</a> resourceProxy.methodNotAvailable(methodName)</h3> <p>Used by the resource method availability checks
to indicate a missing method.</p> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>methodName</td> <td><code>string</code></td></tr></tbody></table> <p><a name="ResourceProxy+get"></a></p> <h3 id="resourceproxy-get-parameters-data-⇒"><a href="#resourceproxy-get-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.get(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourceProxy+list"></a></p> <h3 id="resourceproxy-list-parameters-data-⇒"><a href="#resourceproxy-list-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.list(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourceProxy+create"></a></p> <h3 id="resourceproxy-create-parameters-data-⇒"><a href="#resourceproxy-create-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.create(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourceProxy+replace"></a></p> <h3 id="resourceproxy-replace-parameters-data-⇒"><a href="#resourceproxy-replace-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.replace(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourceProxy+update"></a></p> <h3 id="resourceproxy-update-parameters-data-⇒"><a href="#resourceproxy-update-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.update(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourceProxy+delete"></a></p> <h3 id="resourceproxy-delete-parameters-data-⇒"><a href="#resourceproxy-delete-parameters-data-⇒" class="header-anchor">#</a> resourceProxy.delete(parameters, data) ⇒</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourceProxy"><code>ResourceProxy</code></a><br> <strong>Returns</strong>: Function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>parameters</td> <td><code>Array</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/reference/" class="prev router-link-active">
        Reference (JS)
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.c7d15b50.js" defer></script><script src="/assets/js/2.0dafffee.js" defer></script><script src="/assets/js/47.c5850462.js" defer></script>
  </body>
</html>
