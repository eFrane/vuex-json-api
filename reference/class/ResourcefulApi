<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Json:Api for Vuex</title>
    <meta name="generator" content="VuePress 1.8.0">
    
    <meta name="description" content="Automagic Json:Api integration for Vuex">
    
    <link rel="preload" href="/assets/css/0.styles.ba26daf8.css" as="style"><link rel="preload" href="/assets/js/app.bbfb528b.js" as="script"><link rel="preload" href="/assets/js/2.0dafffee.js" as="script"><link rel="preload" href="/assets/js/42.ac8985e6.js" as="script"><link rel="prefetch" href="/assets/js/10.6f689e45.js"><link rel="prefetch" href="/assets/js/11.3e854a07.js"><link rel="prefetch" href="/assets/js/12.5e6fc63d.js"><link rel="prefetch" href="/assets/js/13.b6b5cdaa.js"><link rel="prefetch" href="/assets/js/14.c2149562.js"><link rel="prefetch" href="/assets/js/15.3b44d0e0.js"><link rel="prefetch" href="/assets/js/16.5963f83e.js"><link rel="prefetch" href="/assets/js/17.e10ac304.js"><link rel="prefetch" href="/assets/js/18.557daba9.js"><link rel="prefetch" href="/assets/js/19.d48ac0dd.js"><link rel="prefetch" href="/assets/js/20.f830c59d.js"><link rel="prefetch" href="/assets/js/21.47a4ed11.js"><link rel="prefetch" href="/assets/js/22.cb5d3132.js"><link rel="prefetch" href="/assets/js/23.7c076f99.js"><link rel="prefetch" href="/assets/js/24.6ac270b7.js"><link rel="prefetch" href="/assets/js/25.4f939962.js"><link rel="prefetch" href="/assets/js/26.0bf0a7fd.js"><link rel="prefetch" href="/assets/js/27.99d4a990.js"><link rel="prefetch" href="/assets/js/28.cdd1dd4b.js"><link rel="prefetch" href="/assets/js/29.4e9281f4.js"><link rel="prefetch" href="/assets/js/3.63095073.js"><link rel="prefetch" href="/assets/js/30.fbd48ebc.js"><link rel="prefetch" href="/assets/js/31.67d4c9e9.js"><link rel="prefetch" href="/assets/js/32.aa9038ac.js"><link rel="prefetch" href="/assets/js/33.228977a9.js"><link rel="prefetch" href="/assets/js/34.8be564c9.js"><link rel="prefetch" href="/assets/js/35.6b3da8ae.js"><link rel="prefetch" href="/assets/js/36.329aa466.js"><link rel="prefetch" href="/assets/js/37.00eac06c.js"><link rel="prefetch" href="/assets/js/38.aed504b0.js"><link rel="prefetch" href="/assets/js/39.1223306c.js"><link rel="prefetch" href="/assets/js/4.bfe88385.js"><link rel="prefetch" href="/assets/js/40.04386673.js"><link rel="prefetch" href="/assets/js/41.15514872.js"><link rel="prefetch" href="/assets/js/43.75e46776.js"><link rel="prefetch" href="/assets/js/44.f73cfb01.js"><link rel="prefetch" href="/assets/js/45.e2c42640.js"><link rel="prefetch" href="/assets/js/46.8c55b03e.js"><link rel="prefetch" href="/assets/js/47.f35fbef8.js"><link rel="prefetch" href="/assets/js/48.8b580e70.js"><link rel="prefetch" href="/assets/js/49.57c4a9c9.js"><link rel="prefetch" href="/assets/js/5.cb249b63.js"><link rel="prefetch" href="/assets/js/50.3fd1e7cb.js"><link rel="prefetch" href="/assets/js/51.4c6b4362.js"><link rel="prefetch" href="/assets/js/52.264f1842.js"><link rel="prefetch" href="/assets/js/53.49f54f43.js"><link rel="prefetch" href="/assets/js/54.4a3a9634.js"><link rel="prefetch" href="/assets/js/55.933fb422.js"><link rel="prefetch" href="/assets/js/56.f2e73cb2.js"><link rel="prefetch" href="/assets/js/57.b9231f84.js"><link rel="prefetch" href="/assets/js/58.83ee0600.js"><link rel="prefetch" href="/assets/js/59.adde428b.js"><link rel="prefetch" href="/assets/js/6.4155846f.js"><link rel="prefetch" href="/assets/js/60.e3dfd946.js"><link rel="prefetch" href="/assets/js/61.f44f8aa3.js"><link rel="prefetch" href="/assets/js/62.1a8322f7.js"><link rel="prefetch" href="/assets/js/63.41a5a998.js"><link rel="prefetch" href="/assets/js/64.7c1cb518.js"><link rel="prefetch" href="/assets/js/65.b037b74d.js"><link rel="prefetch" href="/assets/js/66.a4187ea3.js"><link rel="prefetch" href="/assets/js/67.95fb4f5e.js"><link rel="prefetch" href="/assets/js/7.6ed4aaa9.js"><link rel="prefetch" href="/assets/js/8.58c932af.js"><link rel="prefetch" href="/assets/js/9.9a6f2cf9.js">
    <link rel="stylesheet" href="/assets/css/0.styles.ba26daf8.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container no-sidebar"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><!----> <span class="site-name">Json:Api for Vuex</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <!----> </aside> <main class="page"> <div class="theme-default-content content__default"><p><a name="ResourcefulApi"></a></p> <h2 id="resourcefulapi"><a href="#resourcefulapi" class="header-anchor">#</a> ResourcefulApi</h2> <p><strong>Kind</strong>: global class</p> <ul><li><a href="#ResourcefulApi">ResourcefulApi</a> <ul><li><a href="#ResourcefulApi+doRequest">.doRequest(method, url, params, data, options)</a></li> <li><a href="#ResourcefulApi+preprocessData">.preprocessData(data)</a> ⇒ <code>*</code></li> <li><a href="#ResourcefulApi+setupResourcefulRequests">.setupResourcefulRequests(router)</a></li> <li><a href="#ResourcefulApi+setStore">.setStore(store)</a></li> <li><a href="#ResourcefulApi+setupApiModules">.setupApiModules(apiModulesToRegister)</a></li> <li><a href="#ResourcefulApi+registerModule">.registerModule(moduleName, methods)</a></li> <li><a href="#ResourcefulApi+registerResourceMethods">.registerResourceMethods(routeName, methods)</a></li> <li><a href="#ResourcefulApi+registerApiModule">.registerApiModule(moduleName)</a></li> <li><a href="#ResourcefulApi+getAvailableApiModules">.getAvailableApiModules(onlyUnregistered)</a> ⇒ <code>Array</code></li></ul></li></ul> <p><a name="ResourcefulApi+doRequest"></a></p> <h3 id="resourcefulapi-dorequest-method-url-params-data-options"><a href="#resourcefulapi-dorequest-method-url-params-data-options" class="header-anchor">#</a> resourcefulApi.doRequest(method, url, params, data, options)</h3> <p>Extends <code>Api::doRequest()</code> to handle some data preprocessing.</p> <p>The api modules don't require access to all of the response
and they expect the response data to be normalized.</p> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>method</td> <td><code>String</code></td></tr> <tr><td>url</td> <td><code>String</code></td></tr> <tr><td>params</td> <td><code>Object</code></td></tr> <tr><td>data</td> <td><code>Object</code></td></tr> <tr><td>options</td> <td><code>Object</code></td></tr></tbody></table> <p><a name="ResourcefulApi+preprocessData"></a></p> <h3 id="resourcefulapi-preprocessdata-data-⇒"><a href="#resourcefulapi-preprocessdata-data-⇒" class="header-anchor">#</a> resourcefulApi.preprocessData(data) ⇒ <code>*</code></h3> <p>convert RessourceTypes to uppercase
to follow the json:api spects even if the incoming data is not correct</p> <p>this is just a safety net</p> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th></tr></thead> <tbody><tr><td>data</td></tr></tbody></table> <p><a name="ResourcefulApi+setupResourcefulRequests"></a></p> <h3 id="resourcefulapi-setupresourcefulrequests-router"><a href="#resourcefulapi-setupresourcefulrequests-router" class="header-anchor">#</a> resourcefulApi.setupResourcefulRequests(router)</h3> <p>Prepare the routable requests</p> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>router</td> <td><code>route.Router</code></td></tr></tbody></table> <p><a name="ResourcefulApi+setStore"></a></p> <h3 id="resourcefulapi-setstore-store"><a href="#resourcefulapi-setstore-store" class="header-anchor">#</a> resourcefulApi.setStore(store)</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>store</td> <td><code>Vuex.Store</code></td></tr></tbody></table> <p><a name="ResourcefulApi+setupApiModules"></a></p> <h3 id="resourcefulapi-setupapimodules-apimodulestoregister"><a href="#resourcefulapi-setupapimodules-apimodulestoregister" class="header-anchor">#</a> resourcefulApi.setupApiModules(apiModulesToRegister)</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>apiModulesToRegister</td> <td><code>Array</code></td></tr></tbody></table> <p><a name="ResourcefulApi+registerModule"></a></p> <h3 id="resourcefulapi-registermodule-modulename-methods"><a href="#resourcefulapi-registermodule-modulename-methods" class="header-anchor">#</a> resourcefulApi.registerModule(moduleName, methods)</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>moduleName</td> <td><code>String</code></td></tr> <tr><td>methods</td> <td><code>Route</code></td></tr></tbody></table> <p><a name="ResourcefulApi+registerResourceMethods"></a></p> <h3 id="resourcefulapi-registerresourcemethods-routename-methods"><a href="#resourcefulapi-registerresourcemethods-routename-methods" class="header-anchor">#</a> resourcefulApi.registerResourceMethods(routeName, methods)</h3> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>routeName</td> <td><code>String</code></td></tr> <tr><td>methods</td> <td><code>Route</code></td></tr></tbody></table> <p><a name="ResourcefulApi+registerApiModule"></a></p> <h3 id="resourcefulapi-registerapimodule-modulename"><a href="#resourcefulapi-registerapimodule-modulename" class="header-anchor">#</a> resourcefulApi.registerApiModule(moduleName)</h3> <p>Register an api module</p> <p>After api initialization, this is the way to register
non-default modules.</p> <p>its purpose is to get called from store, where its referenced from the initJsonApiPlugin.
At that point <code>this</code> is the store and not the api object</p> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>moduleName</td> <td><code>String</code></td></tr></tbody></table> <p><a name="ResourcefulApi+getAvailableApiModules"></a></p> <h3 id="resourcefulapi-getavailableapimodules-onlyunregistered-⇒-array"><a href="#resourcefulapi-getavailableapimodules-onlyunregistered-⇒-array" class="header-anchor">#</a> resourcefulApi.getAvailableApiModules(onlyUnregistered) ⇒ <code>Array</code></h3> <p>Get a list of available api modules</p> <p>A module is available if it has defined routing.
If <code>onlyUnregistered</code> is set to false, this list
will also return already registered modules.</p> <p><strong>Kind</strong>: instance method of <a href="#ResourcefulApi"><code>ResourcefulApi</code></a></p> <table><thead><tr><th>Param</th> <th>Type</th> <th>Default</th></tr></thead> <tbody><tr><td>onlyUnregistered</td> <td><code>Boolean</code></td> <td><code>true</code></td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/reference/" class="prev router-link-active">
        Reference (JS)
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.bbfb528b.js" defer></script><script src="/assets/js/2.0dafffee.js" defer></script><script src="/assets/js/42.ac8985e6.js" defer></script>
  </body>
</html>
