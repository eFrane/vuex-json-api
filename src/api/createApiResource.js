export function createApiResource () {
  console.time('api: add method proxies for route ' + routeName)

  for (let methodName in methods) {
    if (methods.hasOwnProperty(methodName)) {
      let route = methods[methodName]

      if (methodName.indexOf('related.') === 0) {
        this.registerRelatedResourceMethod(routeName, methodName, route)
        continue
      }

      this[routeName][methodName] = createApiResourceMethodProxy(this, methodName, route)
    }
  }

  console.timeEnd('api: add method proxies for route ' + routeName)
}
