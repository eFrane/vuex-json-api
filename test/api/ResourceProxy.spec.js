import { ResourcefulApi } from '@/api/ResourcefulApi'
import { ResourceProxy } from '@/api/ResourceProxy'
import { Route } from '@/route/Route'

describe('ResourceProxy', () => {
  let sut

  beforeEach(() => {
    const api = new ResourcefulApi()
    sut = new ResourceProxy(api)
  })

  afterEach(() => {
    sut = null
  })

  it('has routes', () => {
    expect(sut.routes).toBeDefined()
    expect(sut.routes).toStrictEqual({})

    const route = new Route('foo', 'get', '/', [])

    sut.addRoute(route)

    expect(sut.routes).toEqual({
      get: route
    })
  })

  it('throws for unavailable methods', () => {
    function testGet () {
      sut.get()
    }

    expect(testGet).toThrowErrorMatchingSnapshot()
  })

  it('creates a resource proxy for a method if none exists', () => {
    sut.addRoute(new Route('foo', 'get', '/', []))

    expect(sut.proxies).toStrictEqual({})

    sut.getProxyForMethod('get')

    expect(sut.proxies).toMatchSnapshot()
  })

  it('returns a proxy for valid route method', () => {
    sut.addRoute(new Route('foo', 'get', '/', []))

    const methodProxy = sut.get()

    expect(methodProxy).toBeInstanceOf(Function)
  })
})
