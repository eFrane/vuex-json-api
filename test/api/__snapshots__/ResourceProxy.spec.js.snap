// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceProxy creates a resource proxy for a method if none exists 1`] = `
Object {
  "get": [Function],
}
`;

exports[`ResourceProxy throws for unavailable methods 1`] = `"Method get not available"`;
